api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -81,6 +85,16 @@ public void validate(Object obj, Errors errors) {
 		validateHelper(obs, errors, ancestors, true);
 		ValidateUtil.validateFieldLengths(errors, obj.getClass(), "accessionNumber", "valueModifier", "valueComplex",
 		    "comment", "voidReason");
+		try {
+			Encounter encounter = obs.getEncounter();
+			Patient patient = encounter.getPatient();
+			if (obs.getPerson() != patient){
+				throw new ValidationException("Obs.error.patient.id.mismatch");
+			}
+		}catch (Exception e){
```

Comment:
The formatting does not look right here. Did you run mvn clean install before committing?